google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/AddConversionBasedUserList.java

```diff
@@ -0,0 +1,147 @@
+public class AddConversionBasedUserList {
+  private static class AddConversionBasedUserListParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+
+    @Parameter(names = ArgumentNames.CONVERSION_ACTION_ID, required = true)
+    private List<Long> conversionActionIds;
+  }
+
+  public static void main(String[] args) {
+    AddConversionBasedUserListParams params = new AddConversionBasedUserListParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+      params.conversionActionIds =
+          Arrays.asList(
+              Long.parseLong("INSERT_CONVERSION_ACTION_ID_HERE"),
+              Long.parseLong("INSERT_CONVERSION_ACTION_ID_HERE"));
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new AddConversionBasedUserList()
+          .runExample(googleAdsClient, params.customerId, params.conversionActionIds);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer Id.
+   * @param conversionActionIds the Ids of the conversion actions for the basic user list.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(
+      GoogleAdsClient googleAdsClient, long customerId, List<Long> conversionActionIds) {
+    List<UserListActionInfo> userListActionInfoList = new ArrayList<>();
+    for (long conversionActionId : conversionActionIds) {
+      // Creates the UserListActionInfo object for a given conversion action.
```

Comment:
Could you explain the concept here? The purpose of the UserListActionInfo may not be obvious to readers.